core/vil1/vil1_stream_url.cxx

```diff
@@ -226,10 +226,10 @@ vil1_stream_url::vil1_stream_url(char const * url)
   char buffer[4096];
 
   // send HTTP 1.1 request.
-  std::sprintf(buffer, "GET /%s / HTTP/1.1\n", path.c_str());
+  std::snprintf(buffer, sizeof(buffer), "GET /%s / HTTP/1.1\n", path.c_str());
   if (!auth.empty())
-    std::sprintf(buffer + std::strlen(buffer), "Authorization:  Basic %s\n", encode_base64(auth).c_str());
-    //    std::sprintf(buffer+std::strlen(buffer), "Authorization:  user  testuser:testuser\n");
+    std::snprintf(buffer + std::strlen(buffer), sizeof(buffer), "Authorization:  Basic %s\n", encode_base64(auth).c_str());
+    //    std::snprintf(buffer+std::strlen(buffer), sizeof(buffer) "Authorization:  user  testuser:testuser\n");
```

Comment:
@andrei-sandor this isn't quite right... notice the 2nd call does not start at the address of `buffer` but `buffer` plus an offset, therefore the size remaining is less than originally; less by the amount already written.

I think you want:

`std::snprintf(buffer + std::strlen(buffer), sizeof(buffer) - std::strlen(buffer), ...`

Maybe make sure I'm not off-by-1 there by sketching out an example on paper...